src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/RazorSemanticTokenLegendEndpoint.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public class SemanticTokenLegendParams : IRequest<SemanticTokenLegend>, IBaseRequest
+    {
+
+    }
+
+    public class SemanticTokenLegend
+    {
+        private static readonly string[] _tokenTypes = new string[] {
+            "minimizedTagHelperDirectiveAttribute",
```

Comment:
Might be valuable to move them to a shared const somewhere given it is used outside this file.